### PR TITLE
fix: issues with cloud foundry deploy scripts to follow best practices

### PR DESCRIPTION
**Pull request by HarikrishnanBalagopal**

### CODE REVIEW

File: samples/enterprise-app/src/deploy-to-cf.sh [hunk 1]
```diff
@@ -9,119 +9,88 @@ if [[ "$(basename "$PWD")" != 'src' ]] ; then
...
-fi
+cd "$working_dir"/gateway || exit
 SPRING_PROFILES_ACTIVE=dev ./mvnw clean package -P dev
-cf push $appname-gateway
-cd $working_dir/gateway # If the user presses ctrl-c during previous step it will take to the $working_dir
+$CF_CLI_TOOL push "$appname"-gateway \
+    --var M2K_DEMO_CUSTOMERS_URL="http://$appname-customers.mybluemix.net/customers" \
+    --var M2K_DEMO_INVENTORY_URL="http://$appname-inventory.mybluemix.net/products" \
+    --var M2K_DEMO_ORDERS_URL="http://$appname-orders.mybluemix.net/orders"
+cd "$working_dir"/gateway || exit # If the user presses ctrl-c during previous step it will take to the $working_dir
```
Comment (ashokponkumar) [inline samples/enterprise-app/src/deploy-to-cf.sh:61]: We can avoid the m2k demo prefix

File: samples/enterprise-app/src/deploy-to-cf.sh [hunk 2]
```diff
@@ -9,119 +9,88 @@ if [[ "$(basename "$PWD")" != 'src' ]] ; then
...
-    sed -i 's/http:\/\/'"$appname"'-customers.mybluemix.net/http:\/\/customers:8080/g' src/main/resources/application-dev.properties
-    sed -i 's/http:\/\/'"$appname"'-inventory.mybluemix.net/http:\/\/inventory:8080/g' src/main/resources/application-dev.properties
-fi
-echo "\n\n======================================================================="
+
+echo -e "\n\n======================================================================="
 echo "========== Deploying Frontend service to Cloud Foundry ================"
 echo "======================================================================="
+cd "$working_dir"/frontend || exit
+CF_STAGING_TIMEOUT=30 $CF_CLI_TOOL push "$appname"-frontend --var M2K_DEMO_GATEWAY_URL="http://$appname-gateway.mybluemix.net"
```
Comment (ashokponkumar) [inline samples/enterprise-app/src/deploy-to-cf.sh:68]: Lets avoid the m2k demo prefix

**Review by ashokponkumar — CHANGES_REQUESTED** [review]
Few suggestions...

**Review by Akash-Nayak — CHANGES_REQUESTED** [review]
Lets also update the instructions related to manual CF deployment in the README files of the gateway and frontend services.
https://github.com/HarikrishnanBalagopal/move2kube-demos/tree/fix/scripts/samples/enterprise-app/src/gateway#deploying-to-cloud-foundry 
https://github.com/HarikrishnanBalagopal/move2kube-demos/tree/fix/scripts/samples/enterprise-app/src/frontend#deploying-to-cloud-foundry

File: samples/enterprise-app/src/deploy-to-cf.sh [hunk 3]
```diff
@@ -9,119 +9,88 @@ if [[ "$(basename "$PWD")" != 'src' ]] ; then
...
-echo $RANDOM$RANDOM > /dev/null # In zsh, to clear the cached $RANDOM$RANDOM value
-rand1=`echo $RANDOM$RANDOM`
-app=`echo enterprise-app`
-appname=`echo $app-$rand1`
+CF_CLI_TOOL="${CF_CLI_TOOL:-cf}"
+
+working_dir="$PWD"
+echo "$RANDOM$RANDOM" > /dev/null # In zsh, to clear the cached $RANDOM$RANDOM value
+rand1="$RANDOM$RANDOM"
+app='enterprise-app'
```
[the remark this answers is not in the document]
Comment (HarikrishnanBalagopal) [reply samples/enterprise-app/src/deploy-to-cf.sh:17]: to avoid parameter expansion

File: samples/enterprise-app/src/deploy-to-cf.sh (hunk 1 above)
Comment (HarikrishnanBalagopal) [reply samples/enterprise-app/src/deploy-to-cf.sh]: done

File: samples/enterprise-app/src/deploy-to-cf.sh (hunk 2 above)
Comment (HarikrishnanBalagopal) [reply samples/enterprise-app/src/deploy-to-cf.sh]: done

File: samples/enterprise-app/src/deploy-to-ibmcloud-cf.sh [hunk 4]
```diff
@@ -6,124 +6,4 @@
...
-echo "curl http://$appname-gateway.mybluemix.net/products"
-
-echo "\n$appname-frontend:
-Visit http://$appname-frontend.mybluemix.net URL on your browser. Please refresh the page if it appears blank."
-
-echo "\n\n======================================================================="
-echo "======================================================================="
-
-trap - INT
+CF_CLI_TOOL='ibmcloud cf' ./deploy-to-cf.sh
```
Comment (Akash-Nayak) [inline samples/enterprise-app/src/deploy-to-ibmcloud-cf.sh:9]: @HarikrishnanBalagopal, I have tested the deploy-to-ibmcloud-cf.sh script on zsh, it's working fine. But, when I run `echo $appname` after running the script, it returns blank. Can you change it to `CF_CLI_TOOL='ibmcloud cf' source deploy-to-cf.sh` and check if the $appname value is accessible after the running the script?

**Review by Akash-Nayak — CHANGES_REQUESTED** [review]

File: samples/enterprise-app/src/deploy-to-ibmcloud-cf.sh (hunk 4 above)
Comment (HarikrishnanBalagopal) [reply samples/enterprise-app/src/deploy-to-ibmcloud-cf.sh:9]: > Lets also update the instructions related to manual CF deployment in the README files of the gateway and frontend services. https://github.com/HarikrishnanBalagopal/move2kube-demos/tree/fix/scripts/samples/enterprise-app/src/gateway#deploying-to-cloud-foundry https://github.com/HarikrishnanBalagopal/move2kube-demos/tree/fix/scripts/samples/enterprise-app/src/frontend#deploying-to-cloud-foundry

done

Comment (HarikrishnanBalagopal) [reply samples/enterprise-app/src/deploy-to-ibmcloud-cf.sh:9]: > @HarikrishnanBalagopal, I have tested the deploy-to-ibmcloud-cf.sh script on zsh, it's working fine. But, when I run `echo $appname` after running the script, it returns blank. Can you change it to `CF_CLI_TOOL='ibmcloud cf' source deploy-to-cf.sh` and check if the $appname value is accessible after the running the script?

That's on purpose. We don't want to pollute the global namespace with all our variables.
We do print the app name several times and also print the full urls to access the services and the website.

Comment (HarikrishnanBalagopal) [reply samples/enterprise-app/src/deploy-to-ibmcloud-cf.sh:9]: Also if we use `source` then anything declared in the terminal can affect the script as well.

**Review by ashokponkumar — APPROVED** [review]

File: samples/enterprise-app/src/deploy-to-ibmcloud-cf.sh (hunk 4 above)
Comment (Akash-Nayak) [reply samples/enterprise-app/src/deploy-to-ibmcloud-cf.sh]: Sure.

File: samples/enterprise-app/src/frontend/README.md [hunk 5]
```diff
@@ -1,15 +1,25 @@
...
-npm install && npm run start:dev
+```sh
+$ npm install && npm run start:dev
 ```
 
+## Deploying to Cloud Foundry
+
+1. Login to your Cloud Foundry account (`cf login` or `cf login --sso`) if you haven't done so already.
+1. First, deploy the [gateway](https://github.com/konveyor/move2kube-demos/tree/main/samples/enterprise-app/src/gateway#deploying-to-cloud-foundry) service to Cloud Foundry before deploying the `frontend` service.
+1. Using the same `$appname` that was used to deploy the backend and `gateway` services, run the below commands to deploy the `frontend` service to Cloud Foundry.
```
Comment (Akash-Nayak) [inline samples/enterprise-app/src/frontend/README.md:15]: LGTM. One small suggestion. Can you change `run the below commands` to `run the below command`.

Comment (HarikrishnanBalagopal) [reply samples/enterprise-app/src/frontend/README.md]: done

**Review by Akash-Nayak — APPROVED** [review]

File: samples/enterprise-app/src/frontend/README.md (hunk 5 above)
Comment (Akash-Nayak) [reply samples/enterprise-app/src/frontend/README.md]: Thanks Hari.. Nice work!

**Pull request merged**